doc/sphinx/source/developer_guide/modules.rst

```diff
@@ -6,4 +6,462 @@ Modules
+|               +---------------+-------------------------------------------------------------------------+
+|               | raw           | Allows to define a raw makefile section                                 |
++---------------+---------------+-------------------------------------------------------------------------+
+
+
+Starting and Stopping a module
+---------------------------------
+
+Together with the periodic function, the module xml can specify a ``START`` and ``STOP`` function. These are called when
+the module is started or stopped, respectively. The ``autorun`` attribute in the module xml's ``periodic`` element
+controls whether your module is started automatically or manually; you can manually start and stop modules from the GCS
+by going to `Settings -> System -> Modules', selecting ``START`` or ``STOP`` and clicking the green checkmark.
+You can find an example of start and stop functions functions in ``sw/airborne/modules/loggers/file\_logger.c``,
+where they are used to open and close the log file.
+
+If modules are loaded with periodical functions that are not locked, a new tab will automatically appear in the setting
+page of the GCS that allows you to start and stop them.
+
+An other possibility is that any file that includes the header "modules.h" can start or stop the periodic tasks.
+
+
+.. _label-module-header-file:
+
+Module Header File
+---------------------
+
+The module header is located in ``sw/airborne/modules/<module-dir>/<module-header.h>``, and functions like a normal .h
+file. The main difference is that any function or variable that is referenced by an XML file needs to be defined as
+``extern`` so that the compiler can find the definition.
+
+By convention any variable and function that is defined in a module header, especially if used outside of the module by
+another module or XML, should be prefixed with the module name or some other unique identifier to help avoid name
+collision.
+
+
+.. _label-module-source-file:
+
+Module Source File
+--------------------
+
+Modules are stand-alone programs, and hence do not have a `main` function in their source code.
+Instead, the autopilot will regularly call other functions that are part of your module, such as a module periodic
+function. Which functions are called is defined by the module xml file described earlier.
+
+The section `Module XML`_ lists the types of functions you can register in the module xml: ``init``, ``periodic``,
+``event`` and ``datalink``, of which init and periodic are the most common.
+The ``init`` function is called once at startup. You can use this function to initialize important variables of your
+module, or memory intensive structures such as large arrays, or for instance to subscribe to new video frames.
+Once the autopilot is fully initialized, it will enter an infinite loop in which it will continuously read new sensor
+data, feed this to the guidance and stabilization controllers, and send new commands to the Bebop's motors.
```

Comment:
That's oddly specific :smile: 
I think that now, "all is modules", so the infinite really just call all "event" functions, then some "periodic" functions.